homeassistant/components/starlink/time.py

```diff
@@ -0,0 +1,97 @@
+"""Contains time pickers exposed by the Starlink integration."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from datetime import UTC, time
+import math
+
+from homeassistant.components.time import TimeEntity, TimeEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import StarlinkData, StarlinkUpdateCoordinator
+from .entity import StarlinkEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up all time entities for this entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        StarlinkTimeEntity(coordinator, description) for description in TIMES
+    )
+
+
+@dataclass(frozen=True)
+class StarlinkTimeEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    value_fn: Callable[[StarlinkData], time | None]
+    update_fn: Callable[[StarlinkUpdateCoordinator, time], Awaitable[None]]
+    available_fn: Callable[[StarlinkData], bool]
+
+
+@dataclass(frozen=True)
+class StarlinkTimeEntityDescription(
+    TimeEntityDescription, StarlinkTimeEntityDescriptionMixin
+):
+    """Describes a Starlink time entity."""
```

Comment:
There's no need for the mixin class, instead enable `kw_only=True` on the  `StarlinkTimeEntityDescription`

```suggestion
@dataclass(frozen=True)
class StarlinkTimeEntityDescription(TimeEntityDescription):
    """Describes a Starlink time entity."""
    value_fn: Callable[[StarlinkData], time | None]
    update_fn: Callable[[StarlinkUpdateCoordinator, time], Awaitable[None]]
    available_fn: Callable[[StarlinkData], bool]
```

Reply:
Does this apply to other sensors too? I can open a PR to update the rest of that's the case

Reply:
Yes, it applies also to other sensors. @joostlek may have already done that though.